### ISSUE TITLE
Visual Studio  issue

### ISSUE THREAD

**Issue by gtarplyer1** [google/filament#344]
Hi,

I know you don't officially support compiling on MS Visual Studio (2017).  I'm trying it anyway, just for the "fun" of it.  If you feel like helping, great! 

  I'm starting with trying to build the BlueGL library, and it fails to build in BlueGLCoreWindowsImpl.S     One issue was a missing line saying ".model flat" before the ".code"  line.  So I added that.
The next problem is lots of undefined symbols of r11    . Any ideas how I can fix that?

Thank.S

**Comment by bejado** [comment]
Hi, we support using Visual Studio to compile, but you must be using a Clang tool chain. That specific file is an assembly file and it sounds like you’re trying to compile with MSVC which we do not support. How did you generate the VS project?

**Comment by gtarplyer1** [comment]
Nevermind, everything is good.  Looks like I got Filament building under VS using the LLVM toolchain. I'll just stick with LLVM for everything.

Thank you again ! :-)

**Comment by gtarplyer1** [comment]
Looks like I spoke too soon.  Even CLANG is having problems compiling that Assembly file.   BlueGLCoreWindowsImpl.S   ".MODEL must precede this directive" , then the "r11 undefined" error after the .MODEL is added.

**Comment by gtarplyer1** [comment]
I'm using CLANG 7.  Maybe I should be using CLANG 6, like the instructions say.

**Comment by gtarplyer1** [comment]
Still a problem.

**Comment by bejado** [comment]
I can't reproduce that exact error, but my hunch is that you're still using MSVC to try to compile BlueGL because [.MODEL](https://msdn.microsoft.com/en-us/library/ss9fh0d6.aspx) is a Microsoft-specific directive. You should run CMake on the root Filament CMakeLists.txt and ensure you're specifying the correct toolchain with the `-T"LLVM-vs2014"` flag.

As for MSVC and Clang compatibility, your intuition is correct in that mixing binaries compiled with different compilers / options may or may not work. See [MSVC compatability](http://clang.llvm.org/docs/MSVCCompatibility.html) for more information.

**Comment by gtarplyer1** [comment]
ok. I'm using VS 2017, and the toolchain I specified was just "LLVM".    I think it couldn't find the LLVM-vs2014 toolchain because cmake complained, I'll try again.  It didn't complain when I specified "LLVM".  As a side note, I thought it was odd that in the Filament compile instructions, you are using VS 2015, but then specify a LLVM-vs2014 toolchain:  cmake -T"LLVM-vs2014" -G "Visual Studio 14 2015 Win64" ../..
Shouldn't the year version be the same?
  Also,  its not specified what folder the cmake command is run FROM, so the ../..  at the end was  confusing a bit.
